changes.go

```diff
@@ -139,3 +158,29 @@ func handleRelations(add addChangeFunc, relations [][]string, addedServices map[
 func handleUnits(add addChangeFunc, services map[string]*charm.ServiceSpec, addedServices, addedMachines map[string]string) {
 	// TODO frankban: implement this.
 }
+
+// parseEndpoint creates an endpoint from its string representation.
+func parseEndpoint(e string) *endpoint {
```

Comment:
What do you think about making this "resilient" to changes in the bundle format? Maybe this should go into Charm package instead?